cmd/oras/repository/ls.go

```diff
@@ -60,15 +68,34 @@ Example - List the repositories under the registry that include values lexically
 	return cmd
 }
 
+func parseRepoPath(opts *repositoryOptions, arg string) error {
+	path := strings.TrimSuffix(arg, "/")
+	if strings.Contains(path, "/") {
+		reference, err := registry.ParseReference(path)
+		if err != nil || reference.Reference != "" {
+			return err
+		}
+		opts.hostname = reference.Registry
+		opts.namespace = reference.Repository
```

Comment:
```suggestion
		opts.namespace = reference.Repository + "/"
```
so that we don't need to calculate it multiple time in a loop at line 96.

Reply:
changed accordingly.